ydb/docs/en/core/dev/example-app/_includes/example-cpp.md

```diff
@@ -206,174 +185,213 @@ The given code snippet prints the following text to the console at startup:
-
-        DECLARE $seriesId AS Uint64;
-        DECLARE $fromDate AS Uint64;
-        DECLARE $toDate AS Uint64;
-
-        SELECT season_id, episode_id, title, air_date FROM episodes
-        WHERE series_id = $seriesId AND air_date >= $fromDate AND air_date <= $toDate;
-    )", path.c_str());
-
-    auto params2 = session.GetParamsBuilder()
-        .AddParam("$seriesId")
-            .Uint64(seriesId)
-            .Build()
-        .AddParam("$fromDate")
-            .Uint64(fromDate.Days())
-            .Build()
-        .AddParam("$toDate")
-            .Uint64(toDate.Days())
-            .Build()
-        .Build();
-
-    // Execute second query.
-    // Transaction control settings continues active transaction (tx) and
-    // commits it at the end of second query execution.
-    result = session.ExecuteDataQuery(
-        query2,
-        TTxControl::Tx(*tx).CommitTx(),
-        std::move(params2)).GetValueSync();
-
-    if (result.IsSuccess()) {
-        resultSet = result.GetResultSet(0);
-    }
+        // Construct next query based on the results of client logic
+        auto query2 = Sprintf(R"(
+            --!syntax_v1
+            PRAGMA TablePathPrefix("%s");
+
+            DECLARE $seriesId AS Uint64;
+            DECLARE $fromDate AS Uint64;
+            DECLARE $toDate AS Uint64;
+
+            SELECT season_id, episode_id, title, air_date FROM episodes
+            WHERE series_id = $seriesId AND air_date >= $fromDate AND air_date <= $toDate;
+        )", path.c_str());
+
+        auto params2 = TParamsBuilder()
+            .AddParam("$seriesId")
+                .Uint64(seriesId)
+                .Build()
+            .AddParam("$fromDate")
+                .Uint64(fromDate.Days())
+                .Build()
+            .AddParam("$toDate")
+                .Uint64(toDate.Days())
+                .Build()
+            .Build();
+
+        // Execute second query.
+        // Transaction control settings continues active transaction (tx) and
+        // commits it at the end of second query execution.
+        auto result2 = session.ExecuteQuery(
+            query2,
+            TTxControl::Tx(tx->GetId()).CommitTx(),
+            params2).GetValueSync();
+        
+        if (!result2.IsSuccess()) {
+            return result2;
+        }
+        resultSet = result2.GetResultSet(0);
+        return result2;
+    }));
 
-    return result;
+    TResultSetParser parser(*resultSet);
+    Cout << "> MultiStep:" << Endl;
+    while (parser.TryNextRow()) {
+        auto airDate = TInstant::Days(*parser.ColumnParser("air_date").GetOptionalUint64());
+
+        Cout << "Episode " << parser.ColumnParser("episode_id").GetOptionalUint64()
+            << ", Season: " << parser.ColumnParser("season_id").GetOptionalUint64()
+            << ", Title: " << parser.ColumnParser("title").GetOptionalUtf8()
+            << ", Air date: " << airDate.FormatLocalTime("%a %b %d, %Y")
+            << Endl;
+    }
 }
```

Comment:
Among these multiple snippets I kinda lost track of where this lambda that is being retried ends; let's add a comment indicating it? Probably somewhere around here?